src/files-mfs/read-pull-stream.js

```diff
@@ -25,37 +23,24 @@ module.exports = (common, options) => {
 
     after(() => common.teardown())
 
-    it('should not read not found, expect error', (done) => {
+    it('should not read not found, expect error', () => {
       const testDir = `/test-${hat()}`
 
-      pull(
-        ipfs.files.readPullStream(`${testDir}/404`),
-        collect((err) => {
-          expect(err).to.exist()
-          expect(err.message).to.contain('does not exist')
-          done()
-        })
-      )
+      return expect(pullToPromise.any(ipfs.files.readPullStream(`${testDir}/404`))).to.eventually.be.rejected
+        .and.be.an.instanceOf(Error)
+        .and.have.property('message')
+        .that.include('does not exist')
```

Comment:
IMO pull-streams pipelines should not be transformed in promises, you lose flexibility and gain 0 value. Pull-streams should either disappear in the future or be left to work as they were created.

Reply:
It's a fair point, though this method seems to have both brevity and expressiveness, which I like.

Reply:
+ we need to get this merged soon! 😅